x/superfluid/keeper/grpc_query.go

```diff
@@ -304,7 +304,12 @@ func (q Querier) UserSuperfluidPositionsPerConcentratedPoolBreakdown(goCtx conte
 		case cltypes.PositionIdToLockNotFoundError:
 			continue
 		case nil:
+			// If we have hit this logic branch, it means that, at one point, the lockId provided existed. If we fetch it again
+			// and it doesn't exist, that means that the lock has matured.
```

Comment:
>  If we fetch it again and it doesn't exist, that means that the lock has matured.

can there be a case where during refetching someone deleted the immature lock?

Reply:
I cant think of an instance where one could "delete an immature locks", locks are only removed from state when they become mature, so a deleted immature lock would be unexpected behavior.